src/blocks/pacman.rs

```diff
@@ -0,0 +1,53 @@
+use std::time::Duration;
+use std::process::Command;
+
+use block::Block;
+use widgets::text::TextWidget;
+use widget::I3BarWidget;
+
+use serde_json::Value;
+use uuid::Uuid;
+
+
+pub struct Pacman {
+    text: TextWidget,
+    id: String,
+    update_interval: Duration,
+}
+
+impl Pacman {
+    pub fn new(config: Value, theme: Value) -> Pacman {
+        {
+            Pacman {
+                id: Uuid::new_v4().simple().to_string(),
+                update_interval: Duration::new(get_u64_default!(config, "interval", 5), 0),
+                text: TextWidget::new(theme.clone()).with_text(""),
+            }
+        }
+        
+    }
+}
+
+impl Block for Pacman
+{
+    fn update(&mut self) -> Option<Duration> {
+        let output = String::from_utf8(
+            Command::new("sh")
+                .arg("-c")
+                .arg("pacman -Sup 2>/dev/null")
```

Comment:
The command you use to retrieve number of updates requires root, that's a No-go. But there is a way! Look at the guys from bumblebee-status: https://github.com/tobi-wan-kenobi/bumblebee-status/blob/master/bumblebee/modules/pacman.py

They use a bash script and fakeroot:
#!/usr/bin/bash

if ! type -P fakeroot >/dev/null; then
	error 'Cannot find the fakeroot binary.'
	exit 1
fi

if [[ -z $CHECKUPDATES_DB ]]; then
	CHECKUPDATES_DB="${TMPDIR:-/tmp}/checkup-db-${USER}/"
fi

trap 'rm -f $CHECKUPDATES_DB/db.lck' INT TERM EXIT

DBPath="${DBPath:-/var/lib/pacman/}"
eval $(awk -F' *= *' '$1 ~ /DBPath/ { print $1 "=" $2 }' /etc/pacman.conf)

mkdir -p "$CHECKUPDATES_DB"
ln -s "${DBPath}/local" "$CHECKUPDATES_DB" &> /dev/null
fakeroot -- pacman -Sy --dbpath "$CHECKUPDATES_DB" --logfile /dev/null &> /dev/null
fakeroot pacman -Su -p --dbpath "$CHECKUPDATES_DB"

exit 0